### ISSUE TITLE
database is locked - maybe journal related

### ISSUE THREAD

**Issue by aaronfultonnz** [requests-cache/requests-cache#828]
### The problem
I am occasionally (once per month) seeing "sqlite3.OperationalError: database is locked" errors and as per the Troubleshooting page, I'm opening a ticket. 

On the last instance of the error I noticed that there was ah http_cache.sqlite-journal file present so I think that is related. The server had been running stably (no expected shutdowns etc) 

I am running a webapp using wsgi, which has a half-dozen users.

While the database was in a locked state, I used the 'fuse' command to verify that the database was not open by any process (which it was not). 

Upon restarting the web app, the issue resolved, the -journal file was deleted. The http-cache.sqlite file is ~15MB in size.

The issue occurred when attempting to clean the database for expired entries, which I do every time someone logs on.
Stack trace:
```
May 23 22:04:33 ip-172-26-4-62 uwsgi[227678]: Traceback (most recent call last):
May 23 22:04:33 ip-172-26-4-62 uwsgi[227678]:   File "/opt/betaworkflow/./server.py", line 207, in betaworkflow_authenticate
May 23 22:04:33 ip-172-26-4-62 uwsgi[227678]:     requests_session.cache.delete(expired=True)  # Clean cache
May 23 22:04:33 ip-172-26-4-62 uwsgi[227678]:   File "/opt/betaworkflow/venv/lib/python3.10/site-packages/requests_cache/backends/sqlite.py", line 89, in delete
May 23 22:04:33 ip-172-26-4-62 uwsgi[227678]:     self._delete_expired()
May 23 22:04:33 ip-172-26-4-62 uwsgi[227678]:   File "/opt/betaworkflow/venv/lib/python3.10/site-packages/requests_cache/backends/sqlite.py", line 104, in _delete_expired
May 23 22:04:33 ip-172-26-4-62 uwsgi[227678]:     con.execute(
May 23 22:04:33 ip-172-26-4-62 uwsgi[227678]: sqlite3.OperationalError: database is locked

```

### Steps to reproduce the behavior
This is hard to reproduce and only happens very occasionally

### Environment
- requests-cache version: 1.0.1
- Python version: 3.10.10
- Platform:  Ubuntu 20.04 on an AWS lightsail instance

**Comment by JWCook** [comment]
Thank you for the details.

It seems to be failing to handle multiple write operations from multiple threads, although I'm not quite sure why. That leftover `.sqlite-journal` file ([rollback journal](https://sqlite.org/tempfiles.html#rollback_journals)) is just a side effect. That's what SQLite uses to write pending changes within a transaction. The `OperationalError` interrupted the transaction, and then the next time your app started, SQLite used the journal file to rollback the incomplete transaction.

It might take a bit longer without a reproducible example, but I should be able to fix this or at least provide a workaround. Can you show me the exact settings you're using to create your `CachedSession` object?

**Comment by aaronfultonnz** [comment]
Thanks for looking at this.

Here is the relevant section of my app. I'm using the mixin. 


```
from requests import Session
from requests_cache import CacheMixin
from requests_ratelimiter import LimiterMixin


class CachedLimiterSession(CacheMixin, LimiterMixin, Session):
    """
    Session class with caching and rate-limiting behavior. Accepts arguments for both
    LimiterSession and CachedSession.
    """


requests_session = CachedLimiterSession(per_minute=60, max_delay=60, expire_after=300, burst=5, cache_name='http_cache.sqlite')

r = requests_session.put(url, data=xml, headers=headers)

...

def purge_cache_all():
    now = datetime.datetime.now()
    requests_session.cache.delete(older_than=now)
```

**Comment by JWCook** [comment]
I made some changes that may fix this. Try installing from the git repo and let me know if you still encounter the same error.

By the way, this line:
```python
requests_session.cache.delete(older_than=now)
```

May not be doing what you want. That will delete _everything_ in the cache (all items created prior to "now") every time someone logs on. What you may want instead is:
```python
requests_session.cache.delete(expired=True)
```

And run that on your server on some regular schedule, rather than per user.

If you do want to clear the whole cache (and not just expired items), this will be much more efficient:
```python
requests_session.cache.clear()
```

**Comment by aaronfultonnz** [comment]
Thanks!

I'll get this live with my next deployment later this week. Thanks for the tips also.

**Comment by JWCook** [comment]
These changes are now available in v1.1. Let me know if you run into any more problems.
